firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java

```diff
@@ -461,11 +460,12 @@ public void registerPendingWritesTask(TaskCompletionSource<Void> userTask) {
       return;
     }
 
-    if (pendingWritesCallbacks.containsKey(largestPendingBatchId)) {
-      pendingWritesCallbacks.get(largestPendingBatchId).add(userTask);
-    } else {
-      pendingWritesCallbacks.put(largestPendingBatchId, Lists.newArrayList(userTask));
+    if (!pendingWritesCallbacks.containsKey(largestPendingBatchId)) {
+      List<TaskCompletionSource<Void>> tasks = new ArrayList();
+      pendingWritesCallbacks.put(largestPendingBatchId, tasks);
```

Comment:
Nit: You can just inline the call to `new ArrayList`, which would remove all those generic parameters.